vm/src/stdlib/math.rs

```diff
@@ -151,19 +151,45 @@ fn math_trunc(value: PyObjectRef, vm: &VirtualMachine) -> PyResult {
     try_magic_method("__trunc__", vm, &value)
 }
 
+/// Applies ceiling to a float, returning an Integral.
+///
+/// # Arguments
+///
+/// * `value` - Either a float or a python object which implements __ceil__
+/// * `vm` - Represents the python state.
+///
+/// # Remarks
+///
+/// Currently the code is applying ceil to the float using rust's builtin
+/// method and then converts to integral using python float's __int__ method.
+///
 fn math_ceil(value: PyObjectRef, vm: &VirtualMachine) -> PyResult {
     if objtype::isinstance(&value, &vm.ctx.float_type()) {
         let v = objfloat::get_value(&value);
-        Ok(vm.ctx.new_float(v.ceil()))
+        let v = vm.ctx.new_float(v.ceil());
```

Comment:
I’d suggest avoiding to go through Python float and calling magic `__int__` by doing the conversion in Rust and directly returning a Python int:
```rust
let v = v.ceil().to_i64().unwrap(); // this should never fail, since `ceil()` returns integer values
Ok(vm.ctx.new_int(v))
```
or something like that (and the same in `floor`).

Reply:
I wanted to avoid that because there are situations where converting from f64 to i64 loses information. An example of something that works in cpython is:
```python
math.ceil(1e308)
```
cpython returns the correct value. But if f64 is unsafely converted into i64 directly in rust then it
returns an incorrect value of -9223372036854775808. Or it crashes the program with the safe conversion.

I think it would be better to convert the float directly into bigint but I couldn't find a way to pass that back as a result.

Reply:
It looks like [`float.__int__`](https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objfloat.rs#L530-L533) uses [`float.trunc`](https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objfloat.rs#L467-L470) which uses [`objfloat::try_to_bigint`](https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objfloat.rs#L92-L112). I'd recommend making that function public and directly calling it rather than the Python method.

Reply:
Then you can import `num::FromPrimitive` and do:

```rust
let v = BigInt::from_f64(v.ceil()).unwrap();
Ok(vm.ctx.new_int(v))
```

;-)

Reply:
I think that would panic if `v` was `nan` or `inf`, though: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=68ab3dca3b31484e7d5c99ed1c881d1a)

Reply:
You’re right. I just wanted to suggest a standard way of converting to `BigInt` (and instead of unwrapping, proper error propagation could be used), but sure, if there’s already a standard way of `float` → `BigInt` conversion that generates proper Python error in RustPython, it’s better. I didn’t see your comment when writing mine.

Reply:
Okay I updated the code based on coolreader18's suggestion.